sus/assertions/unreachable.h

```diff
@@ -36,12 +34,14 @@
 /// https://en.cppreference.com/w/cpp/utility/unreachable) in C++ which is
 /// Undefined Behaviour if reached. It is closer to [`std::abort`](
 /// https://en.cppreference.com/w/cpp/utility/program/abort) except built on
-/// top of [`sus_panic`]($sus_panic).
+/// top of [`sus::panic`]($sus::panic).
 /// The Subspace library matches the safer behaviour of Rust to avoid confusion
 /// and security bugs when working across languages. Use
-/// [`sus_unreachable_unchecked`]($sus_unreachable_unchecked) to
+/// [`sus::unreachable_unchecked`]($sus::unreachable_unchecked) to
 /// indicate to the compiler the code is not reachable.
-#define sus_unreachable() sus_panic_with_message("entered unreachable code")
+[[noreturn, gnu::always_inline, gnu::nodebug]] inline void unreachable(PanicLocation loc = PanicLocation::current()) {
+  [[clang::always_inline]] sus::panic("entered unreachable code", loc);
```

Comment:
We could add a `_sus_always_inline_call` in inline.h? But is this needed if the function is already annotated as such?